textile/features.textile

```diff
@@ -425,6 +426,28 @@ h3(#rest-encryption). Encryption
 ** @(RSE2a)@ Takes an optional @keyLength@ parameter, which is the length in bits of the key to be generated. If unspecified, this is equal to the default @keyLength@ of the default algorithm: for @AES@, 256 bits.
 ** @(RSE2b)@ Returns (or calls back with, if the language cryptographic randomness primitives are blocking or async) the key as a binary (e.g. a byte array, depending on the language)
 
+h3(#rest-annotations). RestAnnotations
+
+* @(RSAN1)@ @RestAnnotations#publish@ function:
+** @(RSAN1a)@ Takes a first argument of a @Message@ identifier in the form of the string of the @serial@ of the message to be annotated, and a second argument representing the annotation to be published.
+*** @(RSAN1a1)@ In languages where method overloading is supported, the first argument should also accept a @Message@ object (which must contain a populated @serial@ field)
+*** @(RSAN1a2)@ The form of the second argument may accept any language-idiomatic representation (e.g. plain objects in untyped languages), but must allow the user to supply at least the @type@, @clientId@, @name@, @count@, @data@, and @extras@ fields
+*** @(RSAN1a3)@ The SDK must validate that the user supplied a @TAN5b@ @type@. (All other fields are optional)
+*** @(RSAN1a4)@ The SDK may validate that the total size of the annotation or (if publishing an array) annotations, calculated identially to message size per "TO3l8":#TO3l8, exceeds the @maxMessageSize@, then the client library should reject the publish and indicate an error with code 40009
+** @(RSAN1c)@ The SDK must POST a request body consisting of an array with one element, which is the annotation to be published
+*** @(RSAN1c1)@ The @Annotation.action@ must be set to @ANNOTATION_CREATE@
+*** @(RSAN1c2)@ The @Annotation.messageSerial@ must be set to the identifier specified in the first argument to @publish()@
+*** @(RSAN1c3)@ If the user has supplied an @Annotation.data@, that must be encoded (and the @encoding@ set) just as it would be for a @Message@, per @RSL4@
+*** @(RSAN1c4)@ If @idempotentRestPublishing@ (see "TO3n":#TO3n) is enabled and the annotation has an empty @id@ attribute, the SDK should generate a string by base64-encoding a sequence of at least 9 bytes obtained from a source of randomness, append @:0@, and set it as the @Annotation.id@
+*** @(RSAN1c5)@ The body must be encoded to the appropriate format per @RSC8@
+*** @(RSAN1c6)@ The body must be sent as a POST request to the endpoint @/channels/{channelName}/messages/{messageSerial}/annotations@, where @messageSerial@ is the identifier specified in the first argument to @publish()@
```

Comment:
What happens when `POST` request fails, is there custom errorCode/ statusCode for annotations for error returned from server